code/modules/reagents/chemistry/machinery/reagentgrinder.dm

```diff
@@ -176,15 +176,15 @@
 		//Fill machine with a bag!
 		if(istype(I, /obj/item/storage/bag))
 				var/obj/item/storage/bag/B = I
-				for (var/obj/item/reagent_containers/food/snacks/grown/G in B.contents)
+				for (var/obj/item/reagent_containers/food/G in B.contents)
 						B.remove_from_storage(G, src)
 						holdingitems += G
 						if(holdingitems && holdingitems.len >= limit) //Sanity checking so the blender doesn't overfill
 								to_chat(user, "<span class='notice'>You fill the All-In-One grinder to the brim.</span>")
 								break
 
 				if(!I.contents.len)
-						to_chat(user, "<span class='notice'>You empty the plant bag into the All-In-One grinder.</span>")
+						to_chat(user, "<span class='notice'>You empty [I] into the All-In-One grinder.</span>")
 
 				src.updateUsrDialog()
 				return 1
```

Comment:
Should this be `0`? Should it be `0` iff at least one item got transferred?

Reply:
I'm honestly not even sure how that return statement is there and the for loop doesn't terminate after the first item. I think it should be one tab to the left and it should be return TRUE.